### PR TITLE
feat(quic): support TLS password protected keyfile

### PR DESCRIPTION
**Pull request by qzhuyan**
## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` and `.zh.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [x] Schema changes are backward compatible

### CODE REVIEW

File: apps/emqx/src/emqx_listeners.erl [hunk 1]
```diff
@@ -388,7 +388,11 @@ do_start_listener(quic, ListenerName, #{bind := Bind} = Opts) ->
                 ] ++
                     case maps:get(cacertfile, SSLOpts, undefined) of
                         undefined -> [];
-                        CaCertFile -> [{cacertfile, binary_to_list(CaCertFile)}]
+                        CaCertFile -> [{cacertfile, str(CaCertFile)}]
+                    end ++
+                    case maps:get(password, SSLOpts, undefined) of
+                        undefined -> [];
+                        Password -> [{password, str(Password)}]
```
Comment (thalesmg) [inline apps/emqx/src/emqx_listeners.erl:395]: Should we wrap this password here with `emqx_secret:wrap`?  Or esockd would still need to support it, I guess?

File: changes/ce/feat-10077.en.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+QUIC TLS supports password protected certfile.
```
Comment (thalesmg) [inline changes/ce/feat-10077.en.md:1]:
```suggestion
Add support for QUIC TLS password protected certificate file.
```

**Review by thalesmg — APPROVED** [review]

**Review by zmstone — APPROVED** [review]

File: apps/emqx/src/emqx_listeners.erl (hunk 1 above)
Comment (zmstone) [reply apps/emqx/src/emqx_listeners.erl:395]: I guess no, because this option is passed down to quic / ssl lib

Comment (qzhuyan) [reply apps/emqx/src/emqx_listeners.erl:395]: I could add support in quicer, but I don't know from where we should start to wrap it? from emqx_schema ?

File: apps/emqx/test/emqx_common_test_helpers.erl [hunk 3]
```diff
@@ -1073,6 +1081,104 @@ latency_up_proxy(off, Name, ProxyHost, ProxyPort) ->
...
 
 %%-------------------------------------------------------------------------------
+%% TLS certs
+%%-------------------------------------------------------------------------------
+gen_ca(Path, Name) ->
+    %% Generate ca.pem and ca.key which will be used to generate certs
+    %% for hosts server and clients
+    ECKeyFile = filename(Path, "~s-ec.key", [Name]),
+    filelib:ensure_dir(ECKeyFile),
+    os:cmd("openssl ecparam -name secp256r1 > " ++ ECKeyFile),
```
Comment (lafirest) [inline apps/emqx/test/emqx_common_test_helpers.erl:1091]: maybe we could save these codes into a script for other suites to generate certs in ci?

**Review by lafirest — APPROVED** [review]

**Pull request merged**